ai/resource_monitor.py

```
@@ -0,0 +1,224 @@
+            finally:
+                with suppress(Exception):
+                    pynvml.nvmlShutdown()
+
+        if shutil.which("nvidia-smi") is None:
+            return {"available": False, "devices": []}
+
+        try:
+            out = subprocess.run(
+                ["nvidia-smi",
+                 "--query-gpu=index,utilization.gpu,memory.used,memory.total,temperature.gpu",
+                 "--format=csv,noheader,nounits"],
```

Ruff (0.15.11)

[error] Starting a process with a partial executable path

(S607)